codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java

```diff
@@ -204,21 +194,14 @@ private void validateState() {
         checkNotNull(channelAndCodec, "parent channel must be set");
     }
 
-    private static Channel checkRegistered(Channel channel) {
-        if (!channel.isRegistered()) {
-            throw new IllegalArgumentException("The channel must be registered to an eventloop.");
-        }
-        return channel;
-    }
-
     private static ChannelHandler checkSharable(ChannelHandler handler) {
         if (!handler.getClass().isAnnotationPresent(Sharable.class)) {
             throw new IllegalArgumentException("The handler must be Sharable");
         }
         return handler;
     }
 
-    private static class ParentChannelAndMultiplexCodec {
+    static class ParentChannelAndMultiplexCodec {
         private final Channel parentChannel;
```

Comment:
its actually the visibility on the member variables that may be problematic ... make these package private :)